plugins/module_utils/foreman_helper.py

```diff
@@ -1588,6 +1596,14 @@ def _is_resolved(spec, what):
     return False
 
 
+def _scope_key(resource_type, scope):
+    # workaround for https://projects.theforeman.org/issues/31714
+    key = None
+    if resource_type in ['content_views', 'repositories'] and scope == 'lifecycle_environment':
+        key = 'environment'
+    return key
```

Comment:
Is there a way to embed this in `scope` i guess you want it always to be used, right?
I know the logic around this workaround is just :exploding_head: .

Reply:
you mean directly in the `scope_for` func?

Reply:
I guess, if we (optionally) pass it the `res_type` so that it can do the same :exploding_head: logic.

Reply:
Yes, that's where i see it, anyway. :+1:

But having the `res_type` only optionally passed can still produce incorrect results if one omits it. :disappointed:

Reply:
So it seems the scope is not only dependent on what is scoped, but also for what it is scoped.
Maybe we should add the `resource_type` as a mandatory parameter to `scope_for`.

Reply:
Well, it's only really special for LCEs as they don't use their right name in params. I've not seen something like this with any other resource.

Reply:
I added it as an optional param for now. but made the automatic code always pass it.

so now only manual calls have it optional?

Reply:
Yes, it's really a strange edge case. The code looks good for me now.